event-schema/event_schema_test.go

```diff
@@ -1,13 +1,218 @@
-package event_schema_test
+package event_schema
 
 import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"log"
+	"os"
 	"testing"
 
+	"github.com/jeremywohl/flatten"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/ory/dockertest/v3"
+	"github.com/rudderlabs/rudder-server/admin"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	migrator "github.com/rudderlabs/rudder-server/services/sql-migrator"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/testhelper"
+	"github.com/rudderlabs/rudder-server/testhelper/destination"
+	"github.com/rudderlabs/rudder-server/utils/logger"
 )
 
-func TestAlert(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "Alert Suite")
+// TODO: Refactor this TestMain so that it doesn't get called everytime ?
+// TODO: Can we use stretchr's testify suite library ?
+func SetupDB(cleanup *testhelper.Cleanup) *destination.PostgresResource {
+	log.Println("Initialize the database here with the necessary table structures.")
+
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Unable to bring up pool for creating containers")
+	}
+
+	pgResource, err := destination.SetupPostgres(pool, cleanup)
+	if err != nil {
+		log.Fatalf("Unable to setup the postgres container: %s", err)
+	}
+
+	mg := &migrator.Migrator{
+		Handle:                     pgResource.DB,
+		MigrationsTable:            "node_migrations",
+		ShouldForceSetLowerVersion: config.GetBool("SQLMigrator.forceSetLowerVersion", true),
+	}
+
+	err = mg.Migrate("node")
+	if err != nil {
+		log.Fatalf("Unable to run the migrations for the node.")
+	}
+
+	// Load self configuration.
+	config.Load()
+
+	logger.Init() // Initialize the logger
+	Init2()
+	Init()
+
+	// Setup the supporting services like stats and jobsdb
+	stats.Setup()
+	jobsDBInit(pgResource)
+
+	return pgResource
+}
+
+func jobsDBInit(pgResource *destination.PostgresResource) {
+	// Setup env for jobsdb.
+	os.Setenv("JOBS_DB_HOST", pgResource.Host)
+	os.Setenv("JOBS_DB_USER", pgResource.User)
+	os.Setenv("JOBS_DB_PASSWORD", pgResource.Password)
+	os.Setenv("JOBS_DB_DB_NAME", pgResource.Database)
+	os.Setenv("JOBS_DB_PORT", pgResource.Port)
```

Comment:
You should use either `t.Setenv()` if you are using `t *testing.T` or `GinkgoT().Setenv()` if you're using Ginkgo.